record/record.controller.js

```diff
@@ -112,6 +112,40 @@
             });
         };
 
+        vm.canShowSharePopup = function() {
```

Comment:
This exact functionality of matching group-id with the user session is done in four different places (two here, and two in navbar). I think you should create a function and call it in all these four places (two here, and two in navbar). it could be part of the authen.js

```javascript
// where allowedGroups is showAcls in this case, and then in that function you already have access to _session so there's no need to pass it.
whateverFnName: function (allowedGroups) {
  ...
}
```

One thing that I noticed is that in the navbar code, we're properly handling empty session, while here the check is not included.

Reply:
In this case the session being available won't limit the button from showing up, but I should be checking for it before calling `some` on `session.attributes`.

Reply:
Why not? If you define

```
showAcls: {
  "show": ["some-user-group"]
}
```

You only want to show it to a special user group, and therefore the user must be logged in (session must be valid). So I think in both cases you need to check the session. Only if the setting is defined as `["*"]` it should be visible regardless of session value.